dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageChecker.java

```diff
@@ -378,48 +417,12 @@ private boolean validateMethodReferenceByClassLoader(MethodSymbolReference metho
     }
   }
 
-  /**
-   * Returns true if the method reference has a valid referent in the classpath via BCEL API.
-   */
-  private boolean validateMethodReferenceByBcelRepository(MethodSymbolReference methodReference) {
-    String className = methodReference.getTargetClassName();
-    String methodName = methodReference.getMethodName();
-    for (JavaClass javaClass : getClassAndSuperClasses(className)) {
-      for (Method method : javaClass.getMethods()) {
-        if (method.getName().equals(methodName)
-            && method.getSignature().equals(methodReference.getDescriptor())) {
-          return true;
-        }
-      }
-    }
-    return false;
-  }
-
-  /**
-   * Returns true if the method reference has a valid referent in the classpath.
-   */
-  private boolean validateMethodReference(MethodSymbolReference methodReference) {
-    // Attempt 1: Find the class and method via the BCEL synthetic repository in ClassDumper.
-    // BCEL API helps to search availability of (package) private class, constructors and
-    // methods that are inaccessible to Java's reflection API or the class loader.
-
-    // Attempt 2: Find the class and method via the class loader of the input class path
-    // in ClassDumper. Class loaders help to resolve methods defined in Java built-in classes.
-    // TODO(#253): check accessor to verify source class has valid access to the symbol
-    return validateMethodReferenceByBcelRepository(methodReference)
-        || validateMethodReferenceByClassLoader(methodReference);
-  }
-
   /**
    * Returns the target class and its superclasses in order (with {@link Object} last). If any can't
    * be found, the list stops with the previous one.
    */
-  private Iterable<JavaClass> getClassAndSuperClasses(String targetClassName) {
-    try {
-      return SUPERCLASSES.breadthFirst(classDumper.loadJavaClass(targetClassName));
-    } catch (ClassNotFoundException ex) {
-      return ImmutableList.of();
-    }
+  private Iterable<JavaClass> getClassAndSuperClasses(JavaClass targetClass) {
```

Comment:
Taking JavaClass as argument makes it easy for callers to distinguish whether the class is in class path or not.